Implementation/Composing/index-v9.md

```diff
@@ -0,0 +1,519 @@
+
+```csharp
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Core.DependencyInjection;
+using Umbraco.Cms.Core.HealthChecks.Checks.Permissions;
+using Umbraco.Cms.Core.HealthChecks.Checks.Security;
+
+namespace My.Website
+{
+    public class MyComposer: IComposer
+    {
+        public void Compose(IUmbracoBuilder builder)
+        {
+            // Remove all HealthChecks
+            builder.HealthChecks().Clear();
+
+            // Explicitly add back the ones we want to use
+            builder.HealthChecks().Add<FolderAndFilePermissionsCheck>();
+            builder.HealthChecks().Add<ExcessiveHeadersCheck>();
+        }
+    }
+}
+```
+
+
+## Attributes
+Umbraco has some useful C# attributes to decorate your composer classes or Types used in collections, to give you further control on how and when your Composers will 'compose'.
+
+### `[ComposeBefore]` and `[ComposeAfter]`
+A finer-grain mechanism can then be used to refine the order of composition. Each composer can specify that it should compose before or after another composer, using the ComposeBefore and ComposeAfter attributes. For instance:
+
+```csharp
+[ComposeBefore(typeof(ThatOtherComposer))]
+public class ThisComposer : IComposer
+{
+    public void Compose(IUmbracoBuilder builder)
+    {
+    }
+}
+```
+
+`ThisComposer` will 'compose' before `ThatOtherComposer`.
+
+:::warning
+If you create a circular dependency then Umbraco will fail to boot and will report the conflicting/circular dependency.
+:::
+
+### [Weight]
+This attribute is used only for `WeightedCollectionBuilders` (see list above). It specifies an integer ordinal value for each item to be added to the weighted collection which controls their sort order. The weighting attribute is not applied to the Composers.
+
+```csharp
+using System;
+using Umbraco.Core;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Dashboards;
+
+namespace Umbraco.Web.Dashboards
+{
+    [Weight(10)]
+    public class FormsDashboard : IDashboard
+    {
+        public string Alias => "formsInstall";
+        public string[] Sections => new [] { Constants.Applications.Forms };
+        public string View => "views/dashboard/forms/formsdashboardintro.html";
+        public IAccessRule[] AccessRules => Array.Empty<IAccessRule>();
+    }
+}
+```
+
+### [HideFromTypeFinder]
+
+This is used to hide a type from being auto-scanned/added to a collection as in some cases certain items/types may need to be added to a collection manually. For example, a Search package may make it optional whether to replace the 'backoffice search' with an ISearchableTree implementation. Type scanning would make this change automatically at start up if the custom implementation was detected via type scanning. This attribute could hide the class from the scanner.
+
+### [DisableComposer] & [Disable]
+
```

Comment:
Hopefully we can nuke this section, perhaps with a section to talk about alternative approaches